VideoLocker/src/main/java/org/edx/mobile/http/OauthRefreshTokenAuthenticator.java

```diff
@@ -0,0 +1,76 @@
+package org.edx.mobile.http;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.inject.Inject;
+import com.jakewharton.retrofit.Ok3Client;
+
+import org.edx.mobile.authentication.LoginService;
+
+import org.edx.mobile.authentication.AuthResponse;
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.module.prefs.PrefManager;
+import org.edx.mobile.util.Config;
+
+import java.io.IOException;
+
+import okhttp3.Authenticator;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.Route;
+import retrofit.RestAdapter;
+import roboguice.RoboGuice;
+
+/**
+ * Created by cleeedx on 4/8/16.
+ */
+public class OauthRefreshTokenAuthenticator implements Authenticator {
+
+    private final Logger logger = new Logger(getClass().getName());
+    private Context context;
+
+
+    @Inject
+    Config config;
+
+    public OauthRefreshTokenAuthenticator(Context context) {
+        this.context = context;
+        RoboGuice.injectMembers(context, this);
+    }
+
+    @Override
+    public Request authenticate(Route route, final Response response) throws IOException {
+        logger.warn(response.toString());
+
+        if (!response.peekBody(200).string().contains("token_expired")) {
```

Comment:
We get back a very specific error from the server side which is less that 200 chars. I didn't really make this super nice because I wasn't entirely sure how to handle checking for the token expired error message but I'll look this over again.